File: cs/Basic/Client.EventTypes/MachineJobEvent.cs
// Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

namespace EventTypes
{
    using Opc.UaFx;

    [OpcEventType("ns=1;s=MachineJobEventType")]
    public class MachineJobEvent : OpcEvent
    {
        #region ---------- Public constructors ----------

        public MachineJobEvent(IOpcReadOnlyNodeDataStore dataStore)
            : base(dataStore)
        {
        }

        #endregion

        #region ---------- Public properties ----------

        public MachineJob Job
        {
            get => this.DataStore.Get<MachineJob>($"1:{nameof(this.Job)}");
        }

        #endregion
    }
}

File: vb/Basic/Client.EventTypes/MachineJobEvent.vb
'Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

Imports Opc.UaFx

Namespace EventTypes
    <OpcEventType("ns=1;s=MachineJobEventType")>
    Public Class MachineJobEvent
        Inherits OpcEvent

        '---------- Public constructors ----------

        Public Sub New(ByVal dataStore As IOpcReadOnlyNodeDataStore)
            MyBase.New(dataStore)
        End Sub

        '---------- Public properties ----------

        Public ReadOnly Property Job As MachineJob
            Get
                Return Me.DataStore.Get(Of MachineJob)($"1:{NameOf(Me.Job)}")
            End Get
        End Property
    End Class
End Namespace
